components/builder-api-client/src/allow_std_io.rs

```diff
@@ -0,0 +1,149 @@
+//! This is copied almost directly from the `futures` crate [1]. The difference is this
+//! implmentation uses the `AsyncRead` and `AsyncWrite` traits from the `tokio` ecosystem. Once
+//! these traits become part of the standard library the need to duplicate the functionality here
+//! will be obsolete.
```

Comment:
The missing methods should all be in trait implementations (so the methods are not allowed). hese additional methods have to do with vectorization or an initialization feature both of which do not have a `tokio` analogue. I did not implement `AsyncSeek` because I could not come up with a way to implement it given the `tokio` API (and it is not needed).

I added an issue to follow up with our custom implementation [here](https://github.com/habitat-sh/habitat/blob/cd6ce796620a9128621804f938937387a085640f/components/builder-api-client/src/allow_std_io.rs#L7).